library/alloc/src/gc.rs

```diff
@@ -1,3 +1,40 @@
+//! Multi-threaded garbage-collected pointers. 'Gc' stands for 'Garbage
+//! Collected'.
+//!
+//! The type [`Gc<T>`][`Gc`] provides shared ownership of a value of type `T`,
+//! allocated in the heap. [`Gc`] pointers are copyable, with copied [`Gc`]s
+//! pointing to the same allocation in the heap.
+//!
+//! The allocation referenced by a [`Gc`] pointer is dropped by the garbage
```

Comment:
I'm not sure about this paragraph as it can be read as "a Gc is dropped (finalised?) as soon as the last reference to it disappears." Maybe something like "The allocation referenced by a Gc pointer is guaranteed not to be dropped while there are still references to it. When there are no longer any references, the garbage collector will drop it, calling any finalisers (in non-deterministic order) in another thread." ?

Reply:
Yeah I prefer that too. How's this: e4e08292e400aabf77c1cc02ae24a54d89d24e5a